homeassistant/components/sunweg/config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Config flow for Sun WEG integration."""
+from sunweg.api import APIHelper
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import CONF_PLANT_ID, DOMAIN
+
+
+class SunWEGConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow class."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialise sun weg server flow."""
+        self.api: APIHelper = None
+        self.user_id = None
+        self.data: dict = {}
+
+    @callback
+    def _async_show_user_form(self, errors=None) -> FlowResult:
+        """Show the form to the user."""
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): str,
+                vol.Required(CONF_PASSWORD): str,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_user(self, user_input=None) -> FlowResult:
+        """Handle the start of the config flow."""
+        if not user_input:
+            return self._async_show_user_form()
+
+        # Initialise the library with the username & a random id each time it is started
+        self.api = APIHelper(user_input[CONF_USERNAME], user_input[CONF_PASSWORD])
+        login_response = await self.hass.async_add_executor_job(self.api.authenticate)
+
+        if not login_response:
+            return self._async_show_user_form({"base": "invalid_auth"})
+
+        self.data = user_input
+        return await self.async_step_plant()
```

Comment:
instead of using self.data, just pass user_input to async_step_plant

Reply:
`self.data` is to persist authentication information. And the `user_input` inside `async_step_plant` is to receive the plant selection, when needed.